lxc/storage_volume.go

```diff
@@ -1214,7 +1214,13 @@ func (c *cmdStorageVolumeRename) Run(cmd *cobra.Command, args []string) error {
 	if isSnapshot {
 		// Create the storage volume entry
 		vol := api.StorageVolumeSnapshotPost{}
-		vol.Name = shared.ExtractSnapshotName(args[2])
```

Comment:
@stgraber the current lxc rename command help is displayed as follows:

```
Usage:
  lxc storage volume rename [<remote>:]<pool> <old name>[/<old snapshot name>] <new name>[/<new snapshot name>] [flags]
```

When renaming a snapshot, it strongly suggests that the following is possible:

```
Usage:
  lxc storage volume rename default myvol1/mysnap1 myvol2/mysnap2
```

However in the current implementation the `myvol2` is ignored and the snapshot is renamed to `mysnap2` in `myvol1`.

This is because only the new snapshot name is submitted in the POST request as vol.Name.

If this is desired functionality, then should the command's help be changed to:

```
Usage:
  lxc storage volume rename [<remote>:]<pool> <old name>[/<old snapshot name>] <new name>] [flags]
```

Reply:
Yeah, that's probably an issue that was introduce with Rename being a small shim around move. And indeed, rename cannot move a snapshot between volumes.

Reply:
@stgraber would you prefer the command be updated to just take a snapshot name, e.g.

```
lxc storage volume rename default myvol1/mysnap1 mysnap2
```

Or adding a check to ensure that the source parent volume name matches the new parent volume name?